examples/src/main/kotlin/io/ethers/examples/tokenswapwitheventlistening/TokenSwapWithEventListening.kt

```diff
@@ -0,0 +1,117 @@
+package io.ethers.examples.tokenswapwitheventlistening
+
+import UniswapV2Factory
+import UniswapV2Pair
+import UniswapV2Router02
+import io.ethers.core.types.Address
+import io.ethers.core.types.BlockId
+import io.ethers.providers.HttpClient
+import io.ethers.providers.Provider
+import io.ethers.providers.WsClient
+import io.ethers.signers.PrivateKeySigner
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+import kotlinx.cli.required
+import java.lang.Exception
+import java.math.BigInteger
+
+/**
+ * Uniswap V2 ETH to ERC20 swap with listening to UniswapV2Pair Sync event
+ */
+class TokenSwapWithEventListening(
+    privateKey: String,
+    routerAddress: String,
+    wsRpcUrl: String,
+    httpRpcUrl: String,
+    private val wethAddress: String,
+    private val tokenAddress: String,
+    private val ethAmount: BigInteger,
+
+    ) {
+    // Init provider for swap and subscription
+    private val wsClient = WsClient(wsRpcUrl)
+    private val wsProvider = Provider(wsClient)
+    private val httpClient = HttpClient(httpRpcUrl)
+    private val httpProvider = Provider(httpClient)
```

Comment:
No need to define separate providers, just use the WS one. Also, you should use just `provider` for variable name for `Provider`.